sklearn/linear_model/_glm/glm.py

```diff
@@ -285,9 +285,10 @@ def fit(self, X, y, sample_weight=None):
                 jac=True,
                 options={
                     "maxiter": self.max_iter,
+                    "maxls": 40,  # default is 20
                     "iprint": (self.verbose > 0) - 1,
                     "gtol": self.tol,
-                    "ftol": 1e3 * np.finfo(float).eps,
+                    "ftol": 64 * np.finfo(float).eps,  # lbfgs is float64 land.
```

Comment:
What's the justification for the 64 magic number?

Reply:
Yes, magic. It's just smaller than 1e3 which was also just magic. 64 is a power of 2 so ftol will be a number with all bits equal zero except one single bit.

Finding good stopping criteria is just hard. For high precision (small tol), we sometimes have tiny but still significant loss differences, e.g. <~ 1e-14 but maximum |gradient| still larger than tol.

Reply:
I agree the previous `ftol` value was probably too lax.

Reply:
As said earlier: 64 is just a smaller number than the previous value 1e3. I tested lower and higher values. This one passes this new test suite.

Reply:
Since two of us asked the same question maybe it's a sign that this answer should be written as an inline comment (e.g. mentioning that the 64 magic constant was found iteratively to make the test suite pass with the lowest possible tolerance values).

Reply:
Done in https://github.com/scikit-learn/scikit-learn/pull/23619/commits/2fa4397bd198c932a851d63b0feb1313c2264014.